tools/resource-gen/main.go

```diff
@@ -19,6 +19,126 @@ import (
 	_ "github.com/kumahq/kuma/api/mesh/v1alpha1"
 )
 
+// NativeApuTemplate for creating a Kuma resource.
+var NativeApiTemplate = template.Must(template.New("resource").Parse(`
+// Generated by tools/resource-gen
+// Run "make generate" to update this file.
+
+{{ $pkg := printf "%s_proto" .Package }}
+{{ $tk := "` + "`" + `" }}
+
+// nolint:whitespace
+package v1alpha1
+
+import (
+	"github.com/golang/protobuf/proto"
+
+	{{ $pkg }} "github.com/kumahq/kuma/api/{{ .Package }}/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/pkg/model"
+	"github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/pkg/registry"
+
+)
+
+{{range .Resources}}
+{{- if not .SkipNativeGenerate }}
+
+// +kubebuilder:object:root=true
+type {{.ResourceType}} struct {
+	metav1.TypeMeta   {{ $tk }}json:",inline"{{ $tk }}
+	metav1.ObjectMeta {{ $tk }}json:"metadata,omitempty"{{ $tk }}
+
+	Mesh   string                        {{ $tk }}json:"mesh,omitempty"{{ $tk }}
```

Comment:
IIUC, types where `.Global` is true should not have a mesh field.

Reply:
@jpeach MeshInsight has `option (kuma.mesh.resource).global = true;` and a `Mesh` field.

I'll see if I can make this match with the existing options. Maybe `option (kuma.mesh.resource).package = "mesh";`

Reply:
we should also open an issue saying that MeshInsights shouldn't have a Mesh field.

Reply:
IIUC, all the Kubernetes types  have a Mesh field so they can implement the `KubernetesObject` interface.

Reply:
Right but for MeshInsights we can add this method ourselves as a helper like we do for Mesh (and return/set Name).

Reply:
Preserving existing behavior in this PR, will open an issue.